racket/collects/racket/async-channel.rkt

```diff
@@ -147,9 +136,143 @@
+          ac (λ (evt) (values evt (λ (v) (get-proc v))))
+          props)
+   [async-channel-put
+    (λ (async-channel-put)
+      (chaperone-procedure async-channel-put
+                           (λ (ac v) (values ac (put-proc v)))))]
+   [async-channel-put-evt
+    (λ (async-channel-put-evt)
+      (chaperone-procedure async-channel-put-evt
+                           (λ (ac v) (values ac (put-proc v)))))]))
+
+;; Contracts -----------------------------------------------
+
+(define (async-channel/c-name ctc)
+  (define elem-name (contract-name (base-async-channel/c-content ctc)))
+  (apply build-compound-type-name
+         'async-channel/c
+         elem-name
+         '()))
+
+(define (add-async-channel-context blame)
+  (blame-add-context blame "the content of"))
+
+(define (check-async-channel/c ctc val blame)
+  (unless (async-channel? val)
+    (raise-blame-error blame val '(expected "an async channel" given: "~e") val)))
+
+(define (check-async-channel/c-np ctc val blame)
+  (if (async-channel? val)
+      #f
+      (λ (neg-party)
+        (raise-blame-error blame #:missing-party neg-party
+                           val '(expected "an async channel" given: "~e") val))))
+
+(define ((async-channel/c-first-order ctc) val)
+  (define elem-ctc (base-async-channel/c-content ctc))
+  (and (async-channel? val)
+       (contract-first-order-passes? elem-ctc val)))
+
+(define (async-channel/c-stronger? a b)
+  (contract-stronger? (base-async-channel/c-content a) (base-async-channel/c-content b)))
+
+(define ((((ho-val-first-projection impersonate/chaperone-async-channel) ctc) blame) val)
+  (define elem-ctc (base-async-channel/c-content ctc))
+  (define vfp (get/build-val-first-projection elem-ctc))
+  (define async-channel-blame (add-async-channel-context blame))
+  (define pos-elem-proj (vfp async-channel-blame))
+  (define neg-elem-proj (vfp (blame-swap async-channel-blame)))
+  (or (check-async-channel/c-np ctc val blame)
+      (λ (neg-party)
+        (impersonate/chaperone-async-channel 
+         val
+         (λ (v) ((pos-elem-proj v) neg-party))
+         (λ (v) ((neg-elem-proj v) neg-party))
+         impersonator-prop:contracted ctc
+         impersonator-prop:blame (blame-add-missing-party blame neg-party)))))
+
+(define ((((ho-projection impersonate/chaperone-async-channel) ctc) blame) val)
+  (let ([elem-ctc (base-async-channel/c-content ctc)])
+    (let ([pos-elem-proj ((contract-projection elem-ctc) blame)]
+          [neg-elem-proj ((contract-projection elem-ctc) (blame-swap blame))])
+      (check-async-channel/c ctc val blame)
+      (impersonate/chaperone-async-channel val
+                                           (λ (v) (pos-elem-proj v))
+                                           (λ (v) (neg-elem-proj v))
+                                           impersonator-prop:contracted ctc
+                                           impersonator-prop:blame blame))))
+
+(define-struct base-async-channel/c (content))
```

Comment:
@samth ...because the code I copied this from used it and I forgot to update it. I'll change it, thanks.